src/processout/processout.ts

```diff
@@ -342,27 +342,72 @@ module ProcessOut {
          * customer
          * A CardForm may also be provided instead of a card if the fields
          * are hosted by ProcessOut
-         * @param  {Card | CardForm} card
+         * @param  {Card | CardForm | ApplePay | PaymentToken} card
          * @param  {any} data
          * @param  {callback} success
          * @param  {callback} error
          * @return {void}
          */
-        public tokenize(val: Card | CardForm | ApplePay, data: any,
-            success: (token: string)  => void,
-            error:   (err: Exception) => void): void {
+        public tokenize(val: Card | CardForm | ApplePay | PaymentToken, data: any,
+                        success: (token: string)  => void,
+                        error:   (err: Exception) => void): void {
 
             if (val instanceof Card)
                 return this.tokenizeCard(<Card>val, data, success, error);
             if (val instanceof CardForm)
                 return this.tokenizeForm(<CardForm>val, data, success, error);
+            if (val instanceof PaymentToken)
+                return this.tokenizePaymentToken(<PaymentToken>val, success, error);
             if (val instanceof ApplePay)
                 return (<ApplePay>val).tokenize(data, success, error);
 
             throw new Exception("processout-js.invalid-type",
                 "The first parameter had an unknown type/instance. The value must be an instance of either Card, CardForm or ApplePay.");
         }
 
+        /**
+         * tokenizePaymentToken takes the payment token payload
+         * and encodes in base64. Then it creates a ProcessOut
+         * token that can be sent to your server and used to charge your
+         * customer
+         * @param  {PaymentToken} token
+         * @param  {callback} success
+         * @param  {callback} error
+         * @return {void}
+         */
+         protected tokenizePaymentToken(token: PaymentToken,
+                                        success: (token: string) => void,
+                                        error:   (err: Exception) => void): void {
+
+            let data = {}
+            const tokenType = token.getTokenType();
+            
+            switch(tokenType) {
```

Comment:
This `switch` statement has no default case. If the `tokenType` doesn't match one of the covered cases for whatever reason `data` will remain an empty object and the request will be sent anyway. If this isn't expected behaviour I think you should add a default case that throws an exception. If it is expected behaviour I think you should add one anyway so that anybody reading this code knows it is intentional.